crates/autopilot/src/database/events.rs

```diff
@@ -54,25 +54,10 @@ impl EventStoring<ContractEvent> for Postgres {
         block_number.try_into().context("block number is negative")
     }
 
-    async fn append_events(&mut self, events: Vec<EthContractEvent<ContractEvent>>) -> Result<()> {
```

Comment:
We previously used two separate methods so that our Database implementation can be optimized (i.e. we don't do a "delete rows" query when there is no reorg).

Maybe the event handler can choose to call `append_events` vs `replace_events` if there are no reorgs? I don't think this is super critical though and can be done in a follow-up (although it might make sense to not move the method if we intend on making the change).